mapvalidator/keys_are_test.go

```diff
@@ -0,0 +1,119 @@
+package mapvalidator
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/int64validator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+)
+
+func TestKeysAreValidator(t *testing.T) {
+	t.Parallel()
+
+	type testCase struct {
+		val               attr.Value
+		keysAreValidators []tfsdk.AttributeValidator
+		expectError       bool
+	}
+	tests := map[string]testCase{
+		"not Map": {
+			val: types.List{
+				ElemType: types.StringType,
+			},
+			expectError: true,
+		},
+		"Map unknown": {
+			val: types.Map{
+				Unknown:  true,
+				ElemType: types.StringType,
+			},
+			expectError: false,
+		},
+		"Map null": {
+			val: types.Map{
+				Null:     true,
+				ElemType: types.StringType,
+			},
+			expectError: false,
+		},
+		"Map key invalid": {
+			val: types.Map{
+				ElemType: types.StringType,
+				Elems: map[string]attr.Value{
+					"one": types.String{Value: "first"},
+					"two": types.String{Value: "second"},
+				},
+			},
+			keysAreValidators: []tfsdk.AttributeValidator{
+				stringvalidator.LengthAtLeast(4),
+			},
+			expectError: true,
```

Comment:
Nit: At some point it might be nice to refactor these validator tests to test the expected error messages, rather than just being conditional that some error was raised.